lib/chef-dk/policyfile/policyfile_location_specification.rb

```diff
@@ -0,0 +1,74 @@
+require "chef-dk/policyfile_lock"
+require "chef-dk/policyfile/local_lock_fetcher"
+require "chef-dk/policyfile/chef_server_lock_fetcher"
+
+module ChefDK
+  module Policyfile
+    class PolicyfileLocationSpecification
+
+      attr_reader :name
+      attr_reader :source_options
+      attr_reader :storage_config
+      attr_reader :chef_config
+      attr_reader :ui
+
+      LOCATION_TYPES = [:git, :local, :server]
```

Comment:
Probably should remove mention of `:git` until we know we're actually gonna implement.